<?xml version="1.0"?>
<launch>
  <!-- Read me

This launch file is prepared for remote operation of drone by keyboard.  Because of limitations of
docker in terminal emulation, the node teleop_twist_keyboard can't be run directly from this launch
file, but from outside. To run it do the following: Join the solution docker in another terminal
window and run ./subt/scripts/teleop.bash

  -->
  <arg name="robot_name" default="" />
  <param name="robot_name" value="$(arg robot_name)"/>
  <env name="ROSCONSOLE_CONFIG_FILE"  value="$(find robot)/custom_rosconsole.conf"/>

  <!-- Launch your controller nodes -->
  <node name="$(arg robot_name)_proxy" pkg="proxy" type="ros_proxy" args="$(arg robot_name)" output="screen">
	  <!--remap from="$(arg robot_name)/front_scan" to="map_scan"/-->
	  <remap from="$(arg robot_name)/front_rgbd/image_raw/compressed" to="$(arg robot_name)/front/image_raw/compressed"/>
	  <remap from="$(arg robot_name)/cmd_vel" to="$(arg robot_name)/cmd_vel_osgar_intentionally_incorrect" />
	  <remap from="$(arg robot_name)/odom" to="/odom_drone"/>
  </node>


  <node name="drone_height" pkg="robot" type="drone_height.py">
	<remap from="scan_down" to="$(arg robot_name)/bottom_scan" />
	<remap from="scan_up" to="$(arg robot_name)/top_scan" />
	<remap from="cmd_vel_drone" to="$(arg robot_name)/cmd_vel" />
	<remap from="cmd_vel" to="$(arg robot_name)/cmd_vel_osgar"/>
	<remap from="odom" to="$(arg robot_name)/odom" />
  </node> 

</launch>
